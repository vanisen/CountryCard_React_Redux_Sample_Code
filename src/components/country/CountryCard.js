import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import { Paper } from '@material-ui/core';
import CountryCardDialog from './CountryCardDialog';
import CountryCardExpanded from './CountryCardExpanded';

const styles = theme => ({
  paper: {
    marginBottom: 9,
  },
  root: {
    width: '100%',
    backgroundColor: theme.palette.background.paper,
    paddingTop: 6,
    paddingBottom: 6
  },
  flagImg: {
    width: 75,
    height: 50,
  }
});


class CountryCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
    };
  }


  handleClickOpen = () => {
    this.setState({ open: true });
  };

  handleClose = () => {
    this.setState({ open: false });
  };

  render() {
    const { country={}, classes } = this.props;
    return (
      <React.Fragment>
        <Paper className={classes.paper} onClick={this.handleClickOpen}>
          <ListItem className={classes.root} button>
            <img className={classes.flagImg} src={country.flag} alt={country.name} />
            <ListItemText primary={country.name} secondary={country.capital} />
          </ListItem>

        </Paper>
        <CountryCardDialog open={this.state.open} handleClose={this.handleClose}>
          <CountryCardExpanded country={country} handleClose={this.handleClose}/>
        </CountryCardDialog>
      </React.Fragment>
    )
  }
}


CountryCard.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(CountryCard);