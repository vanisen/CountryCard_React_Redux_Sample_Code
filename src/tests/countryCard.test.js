import React from 'react';
import renderer from 'react-test-renderer';
import CountryCard from '../components/country/CountryCard';

it('CountryCard snapshot renders correctly', () => {
    const tree = renderer
      .create(<CountryCard />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });