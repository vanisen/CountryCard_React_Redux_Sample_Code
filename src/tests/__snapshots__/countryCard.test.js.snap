// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryCard snapshot renders correctly 1`] = `
Array [
  <div
    className="MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5 CountryCard-paper-1"
    onClick={[Function]}
>
    <div
        className="MuiButtonBase-root-43 MuiListItem-root-31 MuiListItem-default-34 MuiListItem-gutters-39 MuiListItem-button-40 CountryCard-root-2"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
    >
        <img
            alt={undefined}
            className="CountryCard-flagImg-3"
            src={undefined}
        />
        <div
            className="MuiListItemText-root-46"
        />
        <span
            className="MuiTouchRipple-root-52"
        />
    </div>
</div>,
  <div />,
]
`;
